bindings/java/src/org/sleuthkit/datamodel/TimelineManager.java

```diff
@@ -1023,6 +1023,75 @@ private static int booleanToInt(boolean value) {
 	private static boolean intToBoolean(int value) {
 		return value != 0;
 	}
+	
+	/**
+	 * Returns a list of TimelineEvents for the given filter and time range.
+	 * 
+	 * @param startTime start time in Epoch seconds
+	 * @param endTime end time in Epoch seconds
+	 * @param filter TimelineFilter.RootFilter for filtering data
+	 * 
+	 * @return	A list of TimelineEvents for given parameters, if filter is null 
+	 *			or times are invalid an empty list will be returned.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	public List<TimelineEvent> getEventsForFilter(long startTime, long endTime, TimelineFilter.RootFilter filter) throws TskCoreException{
+		List<TimelineEvent> events = new ArrayList<>();
+		
+		if (filter == null) {
+			return events;
+		}
+		
+		if (endTime < startTime) {
+			return events;
+		}
+
+		//build dynamic parts of query
+        String querySql = "SELECT time, file_obj_id, data_source_obj_id, artifact_id, " // NON-NLS
+                          + "  event_id, " //NON-NLS
+                          + " hash_hit, " //NON-NLS
+                          + " tagged, " //NON-NLS
+                          + " event_type_id, super_type_id, "
+                          + " full_description, med_description, short_description " // NON-NLS
+                          + " FROM " + getAugmentedEventsTablesSQL(filter) // NON-NLS
+                          + " WHERE time >= " + startTime + " AND time < " + endTime + " AND " + getSQLWhere(filter) // NON-NLS
+                          + " ORDER BY time"; // NON-NLS
+		
+		sleuthkitCase.acquireSingleUserCaseReadLock();
+		try (CaseDbConnection con = sleuthkitCase.getConnection();
+				Statement stmt = con.createStatement();
+				ResultSet resultSet = stmt.executeQuery(querySql);) {
+			
+			 while (resultSet.next()) {
+                int eventTypeID = resultSet.getInt("event_type_id");
+				EventType eventType = getEventType(eventTypeID).orElseThrow(()
+						-> new TskCoreException("Error mapping event type id " + eventTypeID + "to EventType."));//NON-NLS
+
+				TimelineEvent event =  new TimelineEvent(
+						resultSet.getLong("event_id"), // NON-NLS
+						resultSet.getLong("data_source_obj_id"), // NON-NLS
+						resultSet.getLong("file_obj_id"), // NON-NLS
+						resultSet.getLong("artifact_id"), // NON-NLS
+						resultSet.getLong("time"), // NON-NLS
+						eventType,
+						resultSet.getString("full_description"), // NON-NLS
+						resultSet.getString("med_description"), // NON-NLS
+						resultSet.getString("short_description"), // NON-NLS
+						resultSet.getInt("hash_hit") != 0, //NON-NLS
+						resultSet.getInt("tagged") != 0);
+				
+				events.add(event);
+            }
+			
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error getting count of events from db: " + querySql, ex); // NON-NLS
```

Comment:
we are getting the events themselves, not just the count here, right? Should fix the log message in that case.